### PR TITLE
2. Update todo (edit and toggle complete)

### PR DESCRIPTION
**Pull request by hhe-wtag**

### CODE REVIEW

**Comment by hhu-wtag** [comment]
Please update the base brach.

File: script.js
Comment (hhu-wtag) [inline script.js:1]: Here's my suggestions for the edit workflow:
We'll have some in component state, let's say `isEditing` and `isDone`.

```js
const todo = {
title: "Some todo",
createdAt: new Date(),
isEditing: false,
isDone: false
}
```
Whenever user clicks on edit button of a todo:
1. Get the id of the todo from the HTML element
2. Find the todo by the id from todos list
3. Toggle, the isEditing to `true`
4. Re-render the whole UI

After this, while creating all the todos, we can have a conditional logic, 
if isEditing:
  showEditState()
else:
  showNormalState()

Same idea goes for done also.

**Review by hhu-wtag — CHANGES_REQUESTED** [review]

File: script.js
Comment (hhe-wtag) [reply script.js:1]: I have addressed the changes kindly review

File: script.js [hunk 1]
```diff
@@ -11,29 +11,92 @@ function handleAddTodo() {
...
   });
 
-  return $newTodo;
+  return $todoText;
+}
+
+function toggleTodoDone(todoId) {
+  todos = todos.map((todo) =>
+    todo.id === todoId ? { ...todo, isCompleted: !todo.isCompleted } : todo
+  );
```
Comment (hhu-wtag) [inline script.js]: Too much happening in one line. Let's break it down.

This'll work due to how reference work in JS
```suggestion
  const todo = todos.find(({ id }) => id === todoId)
  
  if(todo === undefined){
    return
  }
  
  todo.isCompleted = !todo.isCompleted
```

File: script.js [hunk 2]
```diff
@@ -11,29 +11,92 @@ function handleAddTodo() {
...
+    updateTodoTitle(todo.id, e.target.value);
+  });
+
+  return $editInput;
+}
+
+function updateTodoTitle(todoId, newTitle) {
+  todos = todos.map((todo) =>
+    todo.id === todoId ? { ...todo, title: newTitle } : todo
+  );
```
Comment (hhu-wtag) [inline script.js]: ditto. Please simplify this

File: script.js [hunk 3]
```diff
@@ -11,29 +11,92 @@ function handleAddTodo() {
...
+  });
+
+  return $editBtn;
+}
+
+function handleTodoEdit(editingTodo) {
+  todos = todos.map((todo) =>
+    todo.id === editingTodo.id ? { ...todo, isEditing: !todo.isEditing } : todo
+  );
+
```
Comment (hhu-wtag) [inline script.js]: ditto

**Review by hhu-wtag — CHANGES_REQUESTED** [review]

**Review by hhu-wtag — APPROVED** [review]

**Pull request merged**